======================================   
Omics Pipe: An Automated Framework for Next Generation Sequencing Analysis
======================================
 
Introduction
==========
Welcome to the documentation for Omics Pipe! Omics pipe is an open-source, modular 
computational platform that automates ‘best practice’ multi-omics data analysis pipelines 
published in Nature Protocols and other commonly used pipelines, such as GATK.  It currently
automates and provides summary reports for two RNA-seq pipelines, variant calling from whole
exome sequencing (WES), variant calling and copy  number variation analysis from whole genome 
sequencing (WGS), two ChIP-seq pipelines and a custom RNA-seq pipeline for personalized genomic medicine 
reporting.  It also provides automated support for interacting with the The Cancer Genome 
Atlas (TCGA) datasets, including automatic download and processing of the samples in this database.  

:doc:`About Omics Pipe <about>`

Installation
==========

:doc:`Local Cluster <installation>`


:doc:`Amazon Web Services (AWS) <AWS_installation>`


Tutorials
===========

:doc:`Step-by-Step Tutorial <tutorial>`

:doc:`Creating a Custom Pipeline <custom_pipelines>`

:doc:`Creating Custom Modules <adding_modules>`


Online Resources
=================

.. rst-class:: html-plain-table
	
====================== ===================================================
Homepage:              `<http://sulab.org/tools/omics-pipe/>`_
Repository:			`<https://bitbucket.org/sulab/omics_pipe>`_
Online Docs:			`<http://packages.python.org/omics_pipe>`_
Download & PyPI:			`<http://pypi.python.org/pypi/omics_pipe>`_
====================== ===================================================


Site Navigation
==================
* :ref:`genindex`
* :ref:`modindex`
* :ref:`search`

