

<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">


<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    
    <title>Table Of Contents &mdash; Omics Pipe v1.1.3 Documentation</title>
    
    <link rel="stylesheet" href="_static/cloud.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <link rel="stylesheet" href="//fonts.googleapis.com/css?family=Noticia+Text|Open+Sans|Droid+Sans+Mono" type="text/css" />
    
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    './',
        VERSION:     '1.1.3',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true
      };
    </script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <script type="text/javascript" src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <script type="text/javascript" src="_static/jquery.cookie.js"></script>
    <script type="text/javascript" src="_static/cloud.js"></script>
    <link rel="author" title="About these documents" href="about.html" />
    <link rel="copyright" title="Copyright" href="copyright.html" />
    <link rel="top" title="Omics Pipe v1.1.3 Documentation" href="#" />
    <link rel="next" title="Omics Pipe: An Automated Framework for Next Generation Sequencing Analysis" href="index.html" /> 
        <meta name="viewport" content="width=device-width, initial-scale=1">
  </head>
  <body role="document">
    <div class="relbar-top">
        
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li class="right" >
          <a href="#.html" title="Table Of Contents"
             accesskey="C">toc</a> &nbsp; &nbsp;</li>
        <li class="right" >
          <a href="index.html" title="Omics Pipe: An Automated Framework for Next Generation Sequencing Analysis"
             accesskey="N">next</a> &nbsp; &nbsp;</li>
    <li><a href="#">Omics Pipe 1.1.3 Documentation</a> &raquo;</li>
 
      </ul>
    </div>
    </div>
  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body" role="main">
            
  <div class="section" id="table-of-contents">
<h1>Table Of Contents<a class="headerlink" href="#table-of-contents" title="Permalink to this headline">Â¶</a></h1>
<div class="toctree-wrapper compound">
<ul>
<li class="toctree-l1"><a class="reference internal" href="index.html">Omics Pipe</a><ul>
<li class="toctree-l2"><a class="reference internal" href="index.html#introduction">Introduction</a></li>
<li class="toctree-l2"><a class="reference internal" href="index.html#installation">Installation</a></li>
<li class="toctree-l2"><a class="reference internal" href="index.html#tutorials">Tutorials</a></li>
<li class="toctree-l2"><a class="reference internal" href="index.html#online-resources">Online Resources</a></li>
<li class="toctree-l2"><a class="reference internal" href="index.html#site-navigation">Site Navigation</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="about.html">About Omics Pipe</a><ul>
<li class="toctree-l2"><a class="reference internal" href="about.html#available-pipelines">Available Pipelines</a></li>
<li class="toctree-l2"><a class="reference internal" href="about.html#users">Users</a></li>
<li class="toctree-l2"><a class="reference internal" href="about.html#developers">Developers</a></li>
<li class="toctree-l2"><a class="reference internal" href="about.html#contact">Contact</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="installation.html">Using Omics Pipe</a><ul>
<li class="toctree-l2"><a class="reference internal" href="installation.html#requirements">Requirements</a></li>
<li class="toctree-l2"><a class="reference internal" href="installation.html#installation">Installation</a></li>
<li class="toctree-l2"><a class="reference internal" href="installation.html#usage">Usage</a></li>
<li class="toctree-l2"><a class="reference internal" href="installation.html#running-omics-pipe-on-amazon-web-services-aws">Running Omics Pipe on Amazon Web Services (AWS)</a></li>
<li class="toctree-l2"><a class="reference internal" href="installation.html#tutorial">Tutorial</a></li>
<li class="toctree-l2"><a class="reference internal" href="installation.html#version-history">Version history</a></li>
<li class="toctree-l2"><a class="reference internal" href="installation.html#documentation">Documentation</a></li>
<li class="toctree-l2"><a class="reference internal" href="installation.html#questions">Questions</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="AWS_installation.html">OmicsPipe on the Amazon Cloud (AWS EC2) Tutorial</a><ul>
<li class="toctree-l2"><a class="reference internal" href="AWS_installation.html#step-1-create-an-aws-account">Step 1: Create an AWS Account</a></li>
<li class="toctree-l2"><a class="reference internal" href="AWS_installation.html#step-2-mac-or-linux-install-starcluster-and-download-config-plugins">Step 2 (Mac or Linux): Install StarCluster and download config/plugins</a></li>
<li class="toctree-l2"><a class="reference internal" href="AWS_installation.html#step-2-windows-load-the-the-omicspipe-on-aws-docker-image-on-your-machine">Step 2 (Windows): Load the the OmicsPipe on AWS docker image on your machine</a></li>
<li class="toctree-l2"><a class="reference internal" href="AWS_installation.html#step-3-configure-starcluster">Step 3: Configure StarCluster</a></li>
<li class="toctree-l2"><a class="reference internal" href="AWS_installation.html#step-4-create-aws-volumes">Step 4: Create AWS Volumes</a></li>
<li class="toctree-l2"><a class="reference internal" href="AWS_installation.html#step-5-launch-the-cluster">Step 5: Launch the Cluster</a></li>
<li class="toctree-l2"><a class="reference internal" href="AWS_installation.html#step-6-upload-data-to-the-cluster">Step 6: Upload data to the cluster</a></li>
<li class="toctree-l2"><a class="reference internal" href="AWS_installation.html#step-7-run-the-test-pipelines">Step 7: Run the test pipelines</a></li>
<li class="toctree-l2"><a class="reference internal" href="AWS_installation.html#step-8-run-the-pipelines-with-your-own-data">Step 8: Run the pipelines with your own data</a></li>
<li class="toctree-l2"><a class="reference internal" href="AWS_installation.html#installing-extra-software">Installing extra software</a></li>
<li class="toctree-l2"><a class="reference internal" href="AWS_installation.html#to-build-your-own-docker-image">To build your own docker image</a></li>
<li class="toctree-l2"><a class="reference internal" href="AWS_installation.html#add-storage-1tb-to-the-cluster-using-lvm-for-advanced-users">Add storage &gt; 1TB to the cluster using LVM (for advanced users)</a></li>
<li class="toctree-l2"><a class="reference internal" href="AWS_installation.html#add-storage-1tb-to-the-cluster-using-raid-0-for-advanced-users">Add storage &gt; 1TB to the cluster using RAID 0 (for advanced users)</a></li>
<li class="toctree-l2"><a class="reference internal" href="AWS_installation.html#backing-up-your-data-to-s3">Backing up your data to S3</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="tutorial.html">Omics Pipe Tutorial</a><ul>
<li class="toctree-l2"><a class="reference internal" href="tutorial.html#installation">Installation</a></li>
<li class="toctree-l2"><a class="reference internal" href="tutorial.html#before-running-omics-pipe-configuring-parameters-file">Before Running Omics Pipe: Configuring Parameters File</a></li>
<li class="toctree-l2"><a class="reference internal" href="tutorial.html#running-omics-pipe">Running Omics Pipe</a></li>
<li class="toctree-l2"><a class="reference internal" href="tutorial.html#running-omics-pipe-with-the-test-data-and-parameters">Running Omics Pipe with the Test Data and Parameters</a></li>
<li class="toctree-l2"><a class="reference internal" href="tutorial.html#running-omics-pipe-with-your-own-data">Running Omics Pipe with your own data</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="parameter_file.html">Omics Pipe Tutorial &#8211; Configuring the Parameter File</a><ul>
<li class="toctree-l2"><a class="reference internal" href="parameter_file.html#example-omics-pipe-parameter-file">Example Omics Pipe Parameter File</a></li>
<li class="toctree-l2"><a class="reference internal" href="parameter_file.html#explanation-of-variables-in-omics-pipe-parameter-file">Explanation of Variables in Omics Pipe Parameter File</a><ul>
<li class="toctree-l3"><a class="reference internal" href="parameter_file.html#rnaseq-count-based-pipeline">RNAseq Count Based Pipeline</a></li>
<li class="toctree-l3"><a class="reference internal" href="parameter_file.html#rnaseq-tuxedo-pipeline">RNAseq Tuxedo Pipeline</a></li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="custom_pipelines.html">Omics Pipe Tutorial &#8211; Creating a Custom Pipeline Script</a><ul>
<li class="toctree-l2"><a class="reference internal" href="custom_pipelines.html#designing-the-structure-of-the-pipeline">Designing the structure of the pipeline</a></li>
<li class="toctree-l2"><a class="reference internal" href="custom_pipelines.html#creating-the-script">Creating the script</a></li>
<li class="toctree-l2"><a class="reference internal" href="custom_pipelines.html#updating-your-parameters-file">Updating your parameters file</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="adding_modules.html">Omics Pipe Tutorial &#8211; Adding a New Module (Tool)</a><ul>
<li class="toctree-l2"><a class="reference internal" href="adding_modules.html#create-a-bash-script">1.      Create a Bash script</a></li>
<li class="toctree-l2"><a class="reference internal" href="adding_modules.html#create-a-python-module">2.      Create a Python module</a></li>
<li class="toctree-l2"><a class="reference internal" href="adding_modules.html#add-custom-python-module-to-your-custom-pipeline">3.      Add custom Python module to your custom pipeline</a></li>
<li class="toctree-l2"><a class="reference internal" href="adding_modules.html#add-new-parameters-to-parameters-file">4.      Add new parameters to parameters file</a></li>
<li class="toctree-l2"><a class="reference internal" href="adding_modules.html#create-custom-module-directory">5.      Create Custom Module Directory</a></li>
<li class="toctree-l2"><a class="reference internal" href="adding_modules.html#run-your-pipeline">6. Run your pipeline</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="pipelines.html">Omics Pipe Available Pipelines</a><ul>
<li class="toctree-l2"><a class="reference internal" href="pipelines.html#rna-seq-tuxedo">RNA-seq (Tuxedo)</a></li>
<li class="toctree-l2"><a class="reference internal" href="pipelines.html#rna-seq-anders-2013">RNA-seq(Anders 2013)</a></li>
<li class="toctree-l2"><a class="reference internal" href="pipelines.html#whole-exome-sequencing-gatk">Whole Exome Sequencing (GATK)</a></li>
<li class="toctree-l2"><a class="reference internal" href="pipelines.html#whole-genome-sequencing-gatk">Whole Genome Sequencing (GATK)</a></li>
<li class="toctree-l2"><a class="reference internal" href="pipelines.html#whole-genome-sequencing-mutect">Whole Genome Sequencing (MUTECT)</a></li>
<li class="toctree-l2"><a class="reference internal" href="pipelines.html#chip-seq-macs">ChIP-seq (MACS)</a></li>
<li class="toctree-l2"><a class="reference internal" href="pipelines.html#chip-seq-homer">ChIP-seq (HOMER)</a></li>
<li class="toctree-l2"><a class="reference internal" href="pipelines.html#breast-cancer-personalized-genomics-report-rnaseq">Breast Cancer Personalized Genomics Report- RNAseq</a></li>
<li class="toctree-l2"><a class="reference internal" href="pipelines.html#tcga-reanalysis-pipeline-rnaseq">TCGA Reanalysis Pipeline - RNAseq</a></li>
<li class="toctree-l2"><a class="reference internal" href="pipelines.html#tcga-reanalysis-pipeline-rnaseq-counts">TCGA Reanalysis Pipeline - RNAseq Counts</a></li>
<li class="toctree-l2"><a class="reference internal" href="pipelines.html#mirnaseq-counts-anders-2013">miRNAseq Counts (Anders 2013)</a></li>
<li class="toctree-l2"><a class="reference internal" href="pipelines.html#mirnaseq-tuxedo">miRNAseq (Tuxedo)</a></li>
<li class="toctree-l2"><a class="reference internal" href="pipelines.html#all-available-modules">All Available Modules</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="databases.html">Reference Databases Needed</a><ul>
<li class="toctree-l2"><a class="reference internal" href="databases.html#all-pipelines">All Pipelines</a><ul>
<li class="toctree-l3"><a class="reference internal" href="databases.html#genome">Genome</a></li>
<li class="toctree-l3"><a class="reference internal" href="databases.html#reference-annotation-files">Reference Annotation Files</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="databases.html#reference-data-for-cancer-reporting-scripts-rnaseq-cancer-tcga-pipelines">Reference Data for Cancer Reporting Scripts (RNAseq cancer, TCGA pipelines)</a><ul>
<li class="toctree-l3"><a class="reference internal" href="databases.html#in-your-omics-pipe-installation-directory-under-omics-pipe-scripts-reporting-ref-place-the-files">In your omics pipe installation directory under omics_pipe/scripts/reporting/ref place the files.</a></li>
<li class="toctree-l3"><a class="reference internal" href="databases.html#in-your-omics-pipe-installation-directory-under-omics-pipe-scripts-reporting-data-place-the-remaining-files">In your omics pipe installation directory under omics_pipe/scripts/reporting/data place the remaining files.</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="databases.html#references-for-variants-rna-seq-cancer-rna-seq-cancer-tcga-wes-and-wgs-pipelines">References for Variants (RNA-seq cancer, RNA-seq cancer TCGA, WES and WGS pipelines)</a><ul>
<li class="toctree-l3"><a class="reference internal" href="databases.html#you-can-put-these-files-in-any-directory-you-will-point-to-their-location-in-the-parameters-file">You can put these files in any directory. You will point to their location in the parameters file.</a></li>
<li class="toctree-l3"><a class="reference internal" href="databases.html#in-your-omics-pipe-installation-directory-under-omics-pipe-scripts-reporting-ref-place-these-files">In your omics pipe installation directory under omics_pipe/scripts/reporting/ref place these files.</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="databases.html#wes-pipeline">WES Pipeline</a></li>
<li class="toctree-l2"><a class="reference internal" href="databases.html#chip-seq-pipelines">ChIP-seq Pipelines</a></li>
<li class="toctree-l2"><a class="reference internal" href="databases.html#snpir-pipelines-rna-seq-cancer-and-rna-seq-cancer-tcga-pipelines">SNPiR Pipelines (RNA-seq cancer and RNA-seq cancer TCGA pipelines)</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="dependencies.html">Third Party Software Dependencies</a><ul>
<li class="toctree-l2"><a class="reference internal" href="dependencies.html#r-packages-needed">R Packages Needed</a></li>
<li class="toctree-l2"><a class="reference internal" href="dependencies.html#rna-seq-tuxedo">RNA-seq (Tuxedo)</a></li>
<li class="toctree-l2"><a class="reference internal" href="dependencies.html#rna-seq-anders-2013">RNA-seq (Anders 2013)</a></li>
<li class="toctree-l2"><a class="reference internal" href="dependencies.html#whole-exome-sequencing-gatk">Whole Exome Sequencing (GATK)</a></li>
<li class="toctree-l2"><a class="reference internal" href="dependencies.html#whole-genome-sequencing-gatk">Whole Genome Sequencing (GATK)</a></li>
<li class="toctree-l2"><a class="reference internal" href="dependencies.html#whole-genome-sequencing-mutect">Whole Genome Sequencing (MUTECT)</a></li>
<li class="toctree-l2"><a class="reference internal" href="dependencies.html#chip-seq-macs">ChIP-seq (MACS)</a></li>
<li class="toctree-l2"><a class="reference internal" href="dependencies.html#chip-seq-homer">ChIP-seq (HOMER)</a></li>
<li class="toctree-l2"><a class="reference internal" href="dependencies.html#breast-cancer-personalized-genomics-report-rnaseq">Breast Cancer Personalized Genomics Report- RNAseq</a></li>
<li class="toctree-l2"><a class="reference internal" href="dependencies.html#tcga-reanalysis-pipeline-rnaseq">TCGA Reanalysis Pipeline - RNAseq</a></li>
<li class="toctree-l2"><a class="reference internal" href="dependencies.html#tcga-reanalysis-pipeline-rnaseq-counts">TCGA Reanalysis Pipeline - RNAseq Counts</a></li>
<li class="toctree-l2"><a class="reference internal" href="dependencies.html#mirnaseq-counts-anders-2013">miRNAseq Counts (Anders 2013)</a></li>
<li class="toctree-l2"><a class="reference internal" href="dependencies.html#mirnaseq-tuxedo">miRNAseq (Tuxedo)</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="resources.html">System Requirements</a></li>
<li class="toctree-l1"><a class="reference internal" href="RNAseq_Tuxedo_Modules.html">RNA-seq Tuxedo Modules</a><ul>
<li class="toctree-l2"><a class="reference internal" href="RNAseq_Tuxedo_Modules.html#fastqc">FASTQC</a></li>
<li class="toctree-l2"><a class="reference internal" href="RNAseq_Tuxedo_Modules.html#tophat">TopHat</a></li>
<li class="toctree-l2"><a class="reference internal" href="RNAseq_Tuxedo_Modules.html#cufflinks">Cufflinks</a></li>
<li class="toctree-l2"><a class="reference internal" href="RNAseq_Tuxedo_Modules.html#cuffmerge">Cuffmerge</a></li>
<li class="toctree-l2"><a class="reference internal" href="RNAseq_Tuxedo_Modules.html#cuffmergetocompare">Cuffmergetocompare</a></li>
<li class="toctree-l2"><a class="reference internal" href="RNAseq_Tuxedo_Modules.html#cuffdiff">Cuffdiff</a></li>
<li class="toctree-l2"><a class="reference internal" href="RNAseq_Tuxedo_Modules.html#r-summary-report">R Summary Report</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="RNAseq_counts_modules.html">RNA-seq Count Based Modules</a><ul>
<li class="toctree-l2"><a class="reference internal" href="RNAseq_counts_modules.html#fastqc">FASTQC</a></li>
<li class="toctree-l2"><a class="reference internal" href="RNAseq_counts_modules.html#star-aligner">STAR Aligner</a></li>
<li class="toctree-l2"><a class="reference internal" href="RNAseq_counts_modules.html#htseq-count">HTSEQ-count</a></li>
<li class="toctree-l2"><a class="reference internal" href="RNAseq_counts_modules.html#r-summary-report-deseq2">R Summary Report - DESEQ2</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="RNAseq_cancer_modules.html">Breast Cancer Personalized Genomics Report- RNAseq</a><ul>
<li class="toctree-l2"><a class="reference internal" href="RNAseq_cancer_modules.html#fastqc">FASTQC</a></li>
<li class="toctree-l2"><a class="reference internal" href="RNAseq_cancer_modules.html#star-aligner">STAR Aligner</a></li>
<li class="toctree-l2"><a class="reference internal" href="RNAseq_cancer_modules.html#htseq-count">HTSEQ-count</a></li>
<li class="toctree-l2"><a class="reference internal" href="RNAseq_cancer_modules.html#rseqc">RSEQC</a></li>
<li class="toctree-l2"><a class="reference internal" href="RNAseq_cancer_modules.html#fusion-catcher">Fusion Catcher</a></li>
<li class="toctree-l2"><a class="reference internal" href="RNAseq_cancer_modules.html#bwa-snpir">BWA/SNPiR</a><ul>
<li class="toctree-l3"><a class="reference internal" href="RNAseq_cancer_modules.html#bwa">BWA</a></li>
<li class="toctree-l3"><a class="reference internal" href="RNAseq_cancer_modules.html#snpir">SNPiR</a></li>
<li class="toctree-l3"><a class="reference internal" href="RNAseq_cancer_modules.html#filter-variants">Filter Variants</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="RNAseq_cancer_modules.html#intogen">Intogen</a></li>
<li class="toctree-l2"><a class="reference internal" href="RNAseq_cancer_modules.html#oncorep-cancer-report">OncoRep Cancer Report</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="RNAseq_cancer_modules_TCGA.html">TCGA Reanalysis Pipeline - RNAseq</a><ul>
<li class="toctree-l2"><a class="reference internal" href="RNAseq_cancer_modules_TCGA.html#tcga-download">TCGA Download</a></li>
<li class="toctree-l2"><a class="reference internal" href="RNAseq_cancer_modules_TCGA.html#fastqc">FASTQC</a></li>
<li class="toctree-l2"><a class="reference internal" href="RNAseq_cancer_modules_TCGA.html#star-aligner">STAR Aligner</a></li>
<li class="toctree-l2"><a class="reference internal" href="RNAseq_cancer_modules_TCGA.html#htseq-count">HTSEQ-count</a></li>
<li class="toctree-l2"><a class="reference internal" href="RNAseq_cancer_modules_TCGA.html#rseqc">RSEQC</a></li>
<li class="toctree-l2"><a class="reference internal" href="RNAseq_cancer_modules_TCGA.html#fusion-catcher">Fusion Catcher</a></li>
<li class="toctree-l2"><a class="reference internal" href="RNAseq_cancer_modules_TCGA.html#bwa-snpir">BWA/SNPiR</a><ul>
<li class="toctree-l3"><a class="reference internal" href="RNAseq_cancer_modules_TCGA.html#bwa">BWA</a></li>
<li class="toctree-l3"><a class="reference internal" href="RNAseq_cancer_modules_TCGA.html#snpir">SNPiR</a></li>
<li class="toctree-l3"><a class="reference internal" href="RNAseq_cancer_modules_TCGA.html#filter-variants">Filter Variants</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="RNAseq_cancer_modules_TCGA.html#intogen">Intogen</a></li>
<li class="toctree-l2"><a class="reference internal" href="RNAseq_cancer_modules_TCGA.html#oncorep-cancer-report">OncoRep Cancer Report</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="RNAseq_counts_modules_TCGA.html">RNA-seq Count Based Modules- TCGA</a><ul>
<li class="toctree-l2"><a class="reference internal" href="RNAseq_counts_modules_TCGA.html#tcga-download">TCGA Download</a></li>
<li class="toctree-l2"><a class="reference internal" href="RNAseq_counts_modules_TCGA.html#fastqc">FASTQC</a></li>
<li class="toctree-l2"><a class="reference internal" href="RNAseq_counts_modules_TCGA.html#star-aligner">STAR Aligner</a></li>
<li class="toctree-l2"><a class="reference internal" href="RNAseq_counts_modules_TCGA.html#htseq-count">HTSEQ-count</a></li>
<li class="toctree-l2"><a class="reference internal" href="RNAseq_counts_modules_TCGA.html#r-summary-report-deseq2">R Summary Report - DESEQ2</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="miRNAseq_Tuxedo_Modules.html">miRNA-seq Tuxedo Modules</a><ul>
<li class="toctree-l2"><a class="reference internal" href="miRNAseq_Tuxedo_Modules.html#cutadapt">CutAdapt</a></li>
<li class="toctree-l2"><a class="reference internal" href="miRNAseq_Tuxedo_Modules.html#fastq-length-filter">Fastq Length Filter</a></li>
<li class="toctree-l2"><a class="reference internal" href="miRNAseq_Tuxedo_Modules.html#fastqc">FASTQC</a></li>
<li class="toctree-l2"><a class="reference internal" href="miRNAseq_Tuxedo_Modules.html#tophat">TopHat</a></li>
<li class="toctree-l2"><a class="reference internal" href="miRNAseq_Tuxedo_Modules.html#cufflinks">Cufflinks</a></li>
<li class="toctree-l2"><a class="reference internal" href="miRNAseq_Tuxedo_Modules.html#cuffmerge">Cuffmerge</a></li>
<li class="toctree-l2"><a class="reference internal" href="miRNAseq_Tuxedo_Modules.html#cuffmergetocompare">Cuffmergetocompare</a></li>
<li class="toctree-l2"><a class="reference internal" href="miRNAseq_Tuxedo_Modules.html#cuffdiff">Cuffdiff</a></li>
<li class="toctree-l2"><a class="reference internal" href="miRNAseq_Tuxedo_Modules.html#r-summary-report">R Summary Report</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="miRNAseq_counts_modules.html">miRNA-seq Count Based Modules</a><ul>
<li class="toctree-l2"><a class="reference internal" href="miRNAseq_counts_modules.html#cutadapt">CutAdapt</a></li>
<li class="toctree-l2"><a class="reference internal" href="miRNAseq_counts_modules.html#fastq-length-filter">Fastq Length Filter</a></li>
<li class="toctree-l2"><a class="reference internal" href="miRNAseq_counts_modules.html#fastqc">FASTQC</a></li>
<li class="toctree-l2"><a class="reference internal" href="miRNAseq_counts_modules.html#star-aligner">STAR Aligner</a></li>
<li class="toctree-l2"><a class="reference internal" href="miRNAseq_counts_modules.html#htseq">HTSEQ</a></li>
<li class="toctree-l2"><a class="reference internal" href="miRNAseq_counts_modules.html#r-summary-report-deseq2">R Summary Report - DESEQ2</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="ChIPseq_modules_HOMER.html">ChIP-seq Modules &#8211; HOMER</a><ul>
<li class="toctree-l2"><a class="reference internal" href="ChIPseq_modules_HOMER.html#fastqc">FASTQC</a></li>
<li class="toctree-l2"><a class="reference internal" href="ChIPseq_modules_HOMER.html#chip-trim">ChIP trim</a></li>
<li class="toctree-l2"><a class="reference internal" href="ChIPseq_modules_HOMER.html#bowtie">Bowtie</a></li>
<li class="toctree-l2"><a class="reference internal" href="ChIPseq_modules_HOMER.html#read-density-homer">Read Density -HOMER</a></li>
<li class="toctree-l2"><a class="reference internal" href="ChIPseq_modules_HOMER.html#peak-detection-homer">Peak Detection - HOMER</a></li>
<li class="toctree-l2"><a class="reference internal" href="ChIPseq_modules_HOMER.html#peak-annotation-visualization-homer">Peak Annotation &amp; Visualization - HOMER</a></li>
<li class="toctree-l2"><a class="reference internal" href="ChIPseq_modules_HOMER.html#find-motifs-homer">Find Motifs - HOMER</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="ChIPseq_modules_MACS.html">ChIP-seq Modules &#8211; MACS</a><ul>
<li class="toctree-l2"><a class="reference internal" href="ChIPseq_modules_MACS.html#fastqc">FASTQC</a></li>
<li class="toctree-l2"><a class="reference internal" href="ChIPseq_modules_MACS.html#chip-trim">ChIP trim</a></li>
<li class="toctree-l2"><a class="reference internal" href="ChIPseq_modules_MACS.html#bowtie">Bowtie</a></li>
<li class="toctree-l2"><a class="reference internal" href="ChIPseq_modules_MACS.html#macs">MACS</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="WGS_WES_modules.html">Whole Genome and Whole Exome Sequencing Modules</a><ul>
<li class="toctree-l2"><a class="reference internal" href="WGS_WES_modules.html#fastqc">FASTQC</a></li>
<li class="toctree-l2"><a class="reference internal" href="WGS_WES_modules.html#bwa-mem">BWA-MEM</a></li>
<li class="toctree-l2"><a class="reference internal" href="WGS_WES_modules.html#picard-mark-duplicates">PICARD Mark Duplicates</a></li>
<li class="toctree-l2"><a class="reference internal" href="WGS_WES_modules.html#gatk-preprocessing">GATK Preprocessing</a><ul>
<li class="toctree-l3"><a class="reference internal" href="WGS_WES_modules.html#wes">WES</a></li>
<li class="toctree-l3"><a class="reference internal" href="WGS_WES_modules.html#wgs">WGS</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="WGS_WES_modules.html#gatk-variant-discovery">GATK Variant Discovery</a></li>
<li class="toctree-l2"><a class="reference internal" href="WGS_WES_modules.html#gatk-variant-filtering">GATK Variant Filtering</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="mutect_modules.html">Whole Genome Sequencing (MUTECT)</a><ul>
<li class="toctree-l2"><a class="reference internal" href="mutect_modules.html#fastqc">FASTQC</a></li>
<li class="toctree-l2"><a class="reference internal" href="mutect_modules.html#bwa-mem">BWA-MEM</a></li>
<li class="toctree-l2"><a class="reference internal" href="mutect_modules.html#mutect">MUTECT</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="all_modules.html">All Available Modules</a></li>
<li class="toctree-l1"><a class="reference internal" href="history.html">Version History</a><ul>
<li class="toctree-l2"><a class="reference internal" href="history.html#id1"><strong>1.1.3</strong> (2014/08/22)</a></li>
<li class="toctree-l2"><a class="reference internal" href="history.html#b-2014-08-05"><strong>1.1.2b</strong> (2014/08/05)</a><ul>
<li class="toctree-l3"><a class="reference internal" href="history.html#new-features">New Features</a></li>
<li class="toctree-l3"><a class="reference internal" href="history.html#bug-fixes">Bug Fixes</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="history.html#id4"><strong>1.1.0</strong> (2014/07/09)</a></li>
<li class="toctree-l2"><a class="reference internal" href="history.html#id5"><strong>1.0.16</strong> (2014/07/01)</a><ul>
<li class="toctree-l3"><a class="reference internal" href="history.html#id6">New Features</a></li>
<li class="toctree-l3"><a class="reference internal" href="history.html#id7">Bug Fixes</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="history.html#id8"><strong>1.0.15</strong> (2014/06/20)</a><ul>
<li class="toctree-l3"><a class="reference internal" href="history.html#id9">New Features</a></li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="copyright.html">Copyright &amp; License</a><ul>
<li class="toctree-l2"><a class="reference internal" href="copyright.html#omics-pipe">Omics Pipe</a></li>
</ul>
</li>
</ul>
</div>
</div>


          </div>
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
        <p class="logo"><a href="#" title="contents">
          <img class="logo" src="_static/logo.jpg" alt="Logo"/>
        </a></p>
  <div class="sphinxnext">
    <h4>Next page</h4>
    <p class="topless"><a href="index.html"
                          title="Next page">&rarr; Omics Pipe: An Automated Framework for Next Generation Sequencing Analysis</a></p>
  </div>
  <div role="note" aria-label="source link">
    <h3>This Page</h3>
    <ul class="this-page-menu">
      <li><a href="_sources/contents.txt"
            rel="nofollow">Show Source</a></li>
    </ul>
   </div>
<div id="searchbox" style="display: none" role="search">
  <h3>Quick search</h3>
    <form class="search" action="search.html" method="get">
      <input type="text" name="q" />
      <input type="submit" value="Go" />
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
    <p class="searchtip" style="font-size: 90%">
    Enter search terms or a module, class or function name.
    </p>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="relbar-bottom">
        
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             >index</a></li>
        <li class="right" >
          <a href="#.html" title="Table Of Contents"
             >toc</a> &nbsp; &nbsp;</li>
        <li class="right" >
          <a href="index.html" title="Omics Pipe: An Automated Framework for Next Generation Sequencing Analysis"
             >next</a> &nbsp; &nbsp;</li>
    <li><a href="#">Omics Pipe 1.1.3 Documentation</a> &raquo;</li>
 
      </ul>
    </div>
    </div>

    <div class="footer" role="contentinfo">
        &copy; <a href="copyright.html">Copyright</a> 2014-2015 Kathleen Fisch, Ph.D..
      Last updated on Sep 21, 2015.
      Created using <a href="http://sphinx-doc.org/">Sphinx</a> 1.3.1.
    </div>
    <!-- cloud_sptheme 1.4 -->
  </body>
</html>